Mage.Sets/src/mage/cards/a/AlaniaDivergentStorm.java

```diff
@@ -0,0 +1,190 @@
+        paid = false;
+        if (this.getTargets().choose(Outcome.DrawCard, controllerId, source.getSourceId(), source, game)) {
+            Player opponent = game.getPlayer(this.getTargets().get(0).getTargets().get(0));
+            if (opponent == null || !opponent.canRespond()){
+                return false;
+            }
+            paid = opponent.drawCards(1, source, game) != 0;
+        }
+        return paid;
+    }
+
+    @Override
+    public AlaniaDivergentStormCost copy() {
+        return new AlaniaDivergentStormCost(this);
+    }
+}
+
+// Based on MastermindPlumCondition
+enum AlaniaDivergentStormCondition implements Condition {
+    instance;
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Spell spell = (Spell) source.getEffects().get(0).getValue("spellCast");
+        if (spell == null) {
+            return false;
+        }
+        AlaniaDivergentStormWatcher watcher = game.getState().getWatcher(AlaniaDivergentStormWatcher.class);
+        return spell.getId().equals(watcher.getIdOfFirstInstantCast(source.getControllerId())) ||
+                spell.getId().equals(watcher.getIdOfFirstSorceryCast(source.getControllerId())) ||
+                spell.getId().equals(watcher.getIdOfFirstOtterCast(source.getControllerId()));
+    }
+}
+
+// Based on FirstSpellCastThisTurnWatcher
+class AlaniaDivergentStormWatcher extends Watcher {
+
+    private final Map<UUID, UUID> playerFirstInstantCast = new HashMap<>();
+    private final Map<UUID, UUID> playerFirstSorceryCast = new HashMap<>();
+    private final Map<UUID, UUID> playerFirstOtterCast = new HashMap<>();
+
+    public AlaniaDivergentStormWatcher() {
+        super(WatcherScope.GAME);
+    }
+
+    @Override
+    public void watch(GameEvent event, Game game) {
+        if (event.getType() != GameEvent.EventType.SPELL_CAST) {
+            return;
+        }
+        Spell spell = (Spell) game.getObject(event.getTargetId());
+        if (spell == null) {
+            return;
+        }
+        if (spell.getCardType(game).contains(CardType.INSTANT) &&
+                !playerFirstInstantCast.containsKey(spell.getControllerId())) {
+            playerFirstInstantCast.put(spell.getControllerId(), spell.getId());
+        }
+        if (spell.getCardType(game).contains(CardType.SORCERY) &&
+                !playerFirstSorceryCast.containsKey(spell.getControllerId())) {
+            playerFirstSorceryCast.put(spell.getControllerId(), spell.getId());
+        }
+        if (spell.getSubtype(game).contains(SubType.OTTER) &&
+                !playerFirstOtterCast.containsKey(spell.getControllerId())) {
+            playerFirstOtterCast.put(spell.getControllerId(), spell.getId());
+        }
+
+    }
+
+    @Override
+    public void reset() {
+        super.reset();
+        playerFirstInstantCast.clear();
+        playerFirstSorceryCast.clear();
+        playerFirstOtterCast.clear();
+    }
+
+    public UUID getIdOfFirstInstantCast(UUID playerId) {
```

Comment:
you don't have to overimplement separate methods, since they're all going to be checked together anyway, more sensible to just have a single check method that takes the controller id and the spell MOR and returns true/false